packages/editor/src/components/post-sync-status/index.js

```diff
@@ -1,34 +1,110 @@
 /**
  * WordPress dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { PanelRow } from '@wordpress/components';
+import {
+	PanelRow,
+	Modal,
+	Button,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	ToggleControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { ReusableBlocksRenameHint } from '@wordpress/block-editor';
+import { getQueryArgs } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
 import { store as editorStore } from '../../store';
 
 export default function PostSyncStatus() {
-	const { syncStatus, postType } = useSelect( ( select ) => {
-		const { getEditedPostAttribute } = select( editorStore );
+	const { editPost } = useDispatch( editorStore );
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const [ syncType, setSyncType ] = useState( 'fully' );
+
+	const { syncStatus, postType, isNewPost } = useSelect( ( select ) => {
+		const { getEditedPostAttribute, isCleanNewPost } =
+			select( editorStore );
 		return {
 			syncStatus: getEditedPostAttribute( 'wp_pattern_sync_status' ),
 			postType: getEditedPostAttribute( 'type' ),
+			isNewPost: isCleanNewPost(),
 		};
 	}, [] );
+
+	useEffect( () => {
+		if ( isNewPost ) {
+			setIsModalOpen( true );
+		}
+		// We only want the modal to open when the page is first loaded.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	const setSyncStatus = () => {
+		editPost( {
+			meta: {
+				wp_pattern_sync_status:
+					syncType === 'unsynced' ? 'unsynced' : null,
+			},
+		} );
+	};
+
 	if ( postType !== 'wp_block' ) {
 		return null;
 	}
-
-	const isFullySynced = ! syncStatus;
+	const { action } = getQueryArgs( window.location.href );
+	const currentSyncStatus = action === 'edit' ? syncStatus : syncType;
 
 	return (
 		<PanelRow className="edit-post-sync-status">
+			{ isModalOpen && (
+				<Modal
+					title={ __( 'Set pattern sync status' ) }
+					onRequestClose={ () => {
+						setIsModalOpen( false );
+					} }
+					overlayClassName="reusable-blocks-menu-items__convert-modal"
+				>
+					<form
+						onSubmit={ ( event ) => {
+							event.preventDefault();
+							setIsModalOpen( false );
+							setSyncStatus();
+						} }
+					>
+						<VStack spacing="5">
+							<ReusableBlocksRenameHint />
+							<ToggleControl
+								label={ __( 'Synced' ) }
+								help={ __(
```

Comment:
Maybe change the help text to match the state depending on `syncType`?

Reply:
According to @SaxonF  the toggle control standard is to always/only show the positive state help text

Reply:
Fair enough, makes sense 👍🏻 I guess it can be seen as a checkbox in that regard. Cheers!